rholang/examples/linking/v0.2/tests/MakeMintTest.rho

```diff
@@ -0,0 +1,46 @@
+import MakeMint, TestSet in {
+  MakeMint!("mintA") | MakeMint!("mintB") |
+  for(mintA <- @"mintA"; mintB <- @"mintB") {
+    mintA!("makePurse", [100], "aliceAPurse") | mintB!("makePurse", [50], "bobBPurse") |
+    for(@aliceAPurse <- @"aliceAPurse"; @bobBPurse <- @"bobBPurse") {
+      TestSet!(
+        "Purses should be created with the given balance.",
+        [
+          [aliceAPurse, "getBalance", [], 100],
+          [bobBPurse, "getBalance", [], 50]
+        ]
+      ) |
+      @[*TestSet, "after"]!("Purses should be created with the given balance.", {
```

Comment:
So we're using full sentences as identifiers to sequence things? It feels odd but I guess whatever works.

Reply:
Originally when I wrote `TestSet` I wanted something that looked kind of like `scalacheck` and there was no way of writing to `stdout`, so I identified tests by full sentences and had the result of the test (`true`/`false` for pass/fail) be sent over the name created by `@` quoting the sentence. E.g. `@"Purses should be created with the given balance."!(true)`.

I thought that looked really nice in the output because it was easy to see what tests passed. But here I wanted to run the tests in a particular order and the only way to know that a test had completed was to wait for its result to be sent on its sentence. So, yes, I ended up using full sentences as identifiers to sequence on.

I'm open to other suggestions if you think that this is not a natural way of doing things.

Reply:
Na I can't think of something shorter. I just find it odd / interesting.